.github/workflows/docs-ci.yaml

```diff
@@ -0,0 +1,54 @@
+jame: documentation CI
+
+on:
+  workflow_call:
+    inputs:
+      git_tag:
+        type: string
+        default: latest
+        description: The git tag (version) from the calling workflow
+        required: true
+
+jobs:
+  docs-ci:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GH_TOKEN_NVIDIA_CI_CD }}
+      TAG: ${{ inputs.git_tag || 'latest' }}
+      PR_TITLE_PREFIX: "task: update documentation for"
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        token: ${{ secrets.GH_TOKEN_NVIDIA_CI_CD }}  # token must be explicitly set here for push to work in following step
+    - name: Make docs
+      run: |
+        make helm-docs api-docs generate-docs-versions-var doca-driver-build-update
+    - name: Close any existing documentation PRs
+      run: |
+        for pr_number in $(gh pr list --search "$PR_TITLE_PREFIX" --json number --jq ".[].number"); do
+          gh pr close $pr_number
+        done
+    - name: Create PR
+      env:
+        DOWNSTREAM_REPO_OWNER: nvidia-ci-cd
+        DOWNSTREAM_FEATURE_BRANCH: update-docs-for-${{ env.TAG }}
+        UPSTREAM_REPO_OWNER: Mellanox
+        UPSTREAM_DEFAULT_BRANCH: main
+        COMMIT_MESSAGE: ${{ env.PR_TITLE_PREFIX }} ${{ env.TAG }}
+      run: |
+        git config user.name  nvidia-ci-cd
+        git config user.email svc-cloud-orch-gh@nvidia.com
+        gh repo fork --remote --default-branch-only
+        gh repo sync $DOWNSTREAM_REPO_OWNER/${{ github.event.repository.name }} --source $UPSTREAM_REPO_OWNER/${{ github.event.repository.name }} --branch $UPSTREAM_DEFAULT_BRANCH
+
+        git checkout -b $DOWNSTREAM_FEATURE_BRANCH
+        git status
+        git add docs
```

Comment:
What will happen if no changed files?

Reply:
Not tested, but based on the code: the `git commit ...` line should return a non-zero exit code (with output "_nothing to commit, working tree clean_") and the workflow will terminate.

Reply:
Ack

Reply:
should we catch it and terminate cleanly?

Reply:
I would argue that it's an edge case, with a relatively obvious default error message (as shared in my previous comment), and we can leave it as is.
The effort to add it is negligible, but I believe in keeping the code shorter and readable.